### ISSUE TITLE
Totals in tfoot

### ISSUE THREAD

**Issue by 57even** [aslagle/reactive-table#247]
Are there any plans to implement totals? I think it could be really useful to implement either subtotals/running totals or grand totals in a `<tfoot>`.

**Comment by aslagle** [comment]
I don't have any plans to do this. Seems like it would be very dependent on the specific columns, so it would be better to just make things more customizable so you can add your own.

**Comment by 57even** [comment]
Wouldn't it just be to add an option at each field to determine if the column should contain a sum?

    {
      key: 'amount',
      label: 'Total amount',
      totalSum: true,
    }

and with a potential to extend it further into something like what Excel offers:

    {
      key: 'amount',
      label: 'Total amount',
      total: '<sum|count|avg|min|max>',
    }

**Comment by aslagle** [comment]
That could work. `total` could also be a [reduce function](http://underscorejs.org/#reduce) to make it more general. 

I don't think I'll do anything like this unless more people want it, but I'm not opposed to it if someone wants to work on a PR. I think it could be difficult to make it integrate with pagination and server-side collections though, and I would want server-side pagination to still work.

**Comment by 57even** [comment]
But the totals could just be sums of the shown rows at each page instead of sums of all rows. It would be misleading if the sum was way higher than what was shown, and I think it's quite simple just to use the reduce function on only the shown rows.

**Comment by aslagle** [comment]
That's interesting because I see it exactly the opposite way. I would expect the sums to use the whole table and I'd find it misleading if they only used the shown rows. In what kind of a table would it be useful to have a sum of only some of the rows? That's why it's complicated to have a feature like this - different users will have different expectations.

**Comment by roblingle** [comment]
I'd expect a total to sum values from all records matching the filter- some of which might not be on the current page, depending on your `rowsPerPage` setting. :)

In the meantime, is there an easy way to do this myself? If not, I'll probably use [publish-counts](https://github.com/percolatestudio/publish-counts), and whenever I change my custom filter for reactiveTable, I'll adjust that subscription too.

**Comment by aslagle** [comment]
No, there's no way to do anything like this. publish-counts and a custom filter sounds like a good option for now.

**Comment by roblingle** [comment]
[Took a quick stab at it](https://github.com/roblingle/reactive-table/commit/a873bc15e5da80f44da4f9f86e2f010b70b8486d), and I think I'll stick with publish-counts. :) 

I'm still wrapping my head around lower-level publications, but it looks like you'd need a second publication to keep your sums updated, right? One with options like limit, one without.

**Comment by aslagle** [comment]
I think you're on the right track with that but keeping the sums updated would be complicated. In aggregateCursor.observeChanges, you'd need separate functions for added, updated and removed to subtract old values and add new values. I'm also not sure how the options should be set up - it's weird to have them on the server but to support functions they'd have to be.
